lib/edge-runtime/utils.ts

```diff
@@ -24,3 +26,35 @@ export function DBChatMessageToGPT(message: DBChatMessage): ChatGPTMessage {
     content: message.content,
   };
 }
+
+export function removeOldestFunctionCalls(
```

Comment:
Should there be a threshold for how long a function call needs to be if it's to be removed?

imagine you had something like

**oldest**
system prompt
assistant prompt
function call length(10)
function call length(10)
function call length(5000)
...
**newest**

You'd end up shonking the two "10s" which wouldn't really help your cause and would unnecessarily lose information

Reply:
Very good point :+1:

2 thoughts:
1. Very few function calls have such short responses (e.g. 10 tokens). They usually have way-too-long responses, since we're retrofitting to API's which haven't been built with LLMs in mind. E.g. in the companies house conversation yesterday, there were 10 assistant messages and 7 function responses. All of the function responses were longer than all of the assistant messages, with 1 exception (shortest fn call of 175 tokens shorter than longest assistant message of 214 tokens).
2. Trading off between "oldest" and "longest" will be tricky. In your example, what if the earlier function calls have 100 tokens? What if there were 10 more user-assistant messages between the short function responses and the long one? Where do we draw the line?

My proposal:
> Only remove oldest function calls which take up >200 tokens (this will be the vast majority, but will cater to your example)